rust/lance/src/dataset/fragment/session.rs

```diff
@@ -0,0 +1,204 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: Copyright The Lance Authors
+
+use crate::dataset::fragment::{
+    resolve_actual_row_ids, FileFragment, FragReadConfig, FragmentReader,
+};
+use arrow_array::RecordBatch;
+use lance_core::datatypes::Schema;
+use lance_core::Result;
+use std::borrow::Cow;
+use std::sync::Arc;
+
+/// A session of [`FileFragment`]
+///
+/// A [`FragmentSession`] maintains states for better performance in continuous take requests.
```

Comment:
This is the result, but not the purpose of `FragmentSession`. 

I suggest we update the comments to clarify that the following tone:

FragmentSession manages a short-lived session, allowing us to maintain internal states instead of creating a new one each time. This API works well for users making repeated requests over the same projected schema.

(Just for an example)